server/src/main/java/org/elasticsearch/cluster/coordination/Coordinator.java

```diff
@@ -122,6 +122,10 @@ private void handleApplyCommit(ApplyCommitRequest applyCommitRequest) {
     }
 
     PublishWithJoinResponse handlePublishRequest(PublishRequest publishRequest) {
+        // overwrite local node
```

Comment:
I'm not sure why this is needed. As we're using the proper transport layer in our tests, this should be taken care of by `PublicationTransportHandler`?

Reply:
Well spotted, I hadn't realised that the transport layer should have been doing this when deserialising the request. The issue is that `DisruptibleMockTransport` doesn't de/serialize the requests before passing them on, so I've added that in defcc57.